lib/meandro/rules/unused_callbacks.ex

```diff
@@ -26,16 +32,49 @@ defmodule Meandro.Rule.UnusedCallback do
     false
   end
 
-  defp analyze_file(file, _ast) do
-    [set_result(file, nil, nil, nil)]
+  defp analyze_file(file, ast) do
+    {_, callbacks} = Macro.prewalk(ast, [], &collect_callbacks/2)
+
+    for {name, arity, line} <- callbacks, is_unused(name, arity, ast) do
+      %Meandro.Rule{
+        file: file,
+        line: line,
+        text: "Callback #{name}/#{arity} is not used anywhere in the module",
+        pattern: {name, arity}
+      }
+    end
+  end
+
+  defp collect_callbacks({:callback, _, _} = callback, acc),
+    do: {callback, [parse(callback) | acc]}
+
+  defp collect_callbacks(other, acc), do: {other, acc}
+
+  defp parse({:callback, meta, [{:"::", _, [definition, _result]}]}) do
+    line = Keyword.get(meta, :line, 0)
+    {name, _, params} = definition
+    {name, length(params), line}
   end
 
-  defp set_result(file, line, callback, arity) do
-    %{
-      file: file,
-      line: line,
-      text: "Callback #{callback}/#{arity} is not used anywhere in the module",
-      pattern: {callback, arity}
-    }
+  defp is_unused(name, arity, ast) do
```

Comment:
let's follow Elixir's standard:
```suggestion
  defp is_unused?(name, arity, ast) do
```

Reply:
👌🏻